service/topology_state_machine.hh

```diff
@@ -127,6 +128,8 @@ struct raft_topology_cmd {
           fence_old_reads  // wait for all reads started before to complete
       };
       command cmd;
+
+      raft_topology_cmd(command c) : cmd(c) {}
```

Comment:
Explicit? But then we do not need it.

Reply:
Being implicit is the entire purpose so I can pass e.g. `raft_topology_cmd::command::barrier` instead of `raft_topology_cmd{raft_topology_cmd::command::barrier}` to functions that take `const raft_topology_cmd&`

Reply:
Such casts may cause surprises. May be not for the rarely used type like here, but we do not save a lot of typing with explicit cast anyway. We also have it in all other places.

Reply:
I really don't like the redundant typing.
How about I extract the constructor to a separate commit and change the code to use it everywhere

Reply:
One may say it is a redundant typing, but the other may note that this is type safety. I know you like type safety as well. But OK, this is internal type with narrow use. We will manage.